File: 0011. Container With Most Water/ContainerWithMostWater.py
class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        result = 0
        l = 0
        r = len(height) - 1
        while l < r:
        	result = max(result, min(height[l], height[r]) * (r - l))
        	if height[l] < height[r]:
        		l += 1
        	else:
        		r -= 1
        return result

File: 0011. Container With Most Water/ContainerWithMostWater.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    var result = 0;
    var l = 0;
    var r = height.length - 1;
    while (l < r) {
    	result = Math.max(result, Math.min(height[l], height[r]) * (r - l));
    	if (height[l] < height[r]) {
    		l++;
    	} else {
    		r--;
    	}
    }
    return result;
};
